src/processRequest.ts

```diff
@@ -349,75 +333,38 @@ export default function processRequest(
- * @param {ProcessRequestOptions} [options] Options.
- * @returns {Promise<
- *   { [key: string]: unknown } | Array<{ [key: string]: unknown }>
- * >} GraphQL operation or batch of operations for a GraphQL server to consume
- *   (usually as the request body). A GraphQL operation typically has the
- *   properties `query` and `variables`.
  */
+export type FileUpload = {
+  filename: string;                                     // File name.
+  mimetype: string;                                     // File MIME type. Provided by the client and can’t be trusted.
+  encoding: string;                                     // File stream transfer encoding.
+  capacitor: WriteStream;                              // A private implementation detail that shouldn’t be used outside `graphql-upload`.
+  createReadStream: FileUploadCreateReadStream;        // Creates a Node.js readable stream of the file’s contents, for processing and storage.
+};
+
+export type FileUploadCreateReadStream = (
+  options?: FileUploadCreateReadStreamOptions // Options (optional)
+) => Readable; // Returns a Node.js readable stream of the file’s contents.
+
+export type FileUploadCreateReadStreamOptions = {
+  options?: {
+    encoding?: ReadStreamOptions["encoding"];          // Specify an encoding for the `data` chunks to be strings (without splitting multi-byte characters across chunks) instead of Node.js `Buffer` instances.
+                                                       // Supported values depend on the `Buffer` implementation and include `utf8`, `ucs2`, `utf16le`, `latin1`, `ascii`, `base64`, `base64url`, or `hex`. Defaults to `utf8`.
+    highWaterMark?: ReadStreamOptions["highWaterMark"];// Maximum number of bytes to store in the internal buffer before ceasing to read from the underlying resource. Defaults to `16384`.
+  };
+};
 
 /**
- * {@linkcode ProcessRequestFunction} options.
- * @typedef {object} ProcessRequestOptions
- * @prop {number} [maxFieldSize] Maximum allowed non file multipart form field
- *   size in bytes; enough for your queries. Defaults to `1000000` (1 MB).
- * @prop {number} [maxFileSize] Maximum allowed file size in bytes. Defaults to
- *   `Infinity`.
- * @prop {number} [maxFiles] Maximum allowed number of files. Defaults to
- *   `Infinity`.
+ * Processes an incoming [GraphQL multipart request](https://github.com/jaydenseric/graphql-multipart-request-spec).
  */
+export type ProcessRequestFunction = (
+  request: IncomingMessage,              // Node.js HTTP server request instance
+  response: ServerResponse,              // Node.js HTTP server response instance
+  options?: ProcessRequestOptions        // Options (optional)
+) => Promise<{ [key: string]: unknown } | Array<{ [key: string]: unknown }>>;
+
+export type ProcessRequestOptions = {
+  maxFieldSize?: number;   // Maximum allowed non-file multipart form field size in bytes; enough for your queries. Defaults to `1000000` (1 MB).
```

Comment:
You have to use JSDoc comments (starting with `/**`) for them to show up in intellisense.

Reply:
Was moving fast just to get things working. Chat Jippity did this bit of code